### PR TITLE
Add icecream-cpp recipe.

### PR DESCRIPTION
**Pull request by renatoGarcia**
Specify library name and version:  **icecream-cpp/0.3.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/icecream-cpp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    url = "https://github.com/renatoGarcia/icecream-cpp"
+    description = "A little library to help with the print debugging on C++11 and forward."
+    topics = ("debug", "single-header-lib", "print")
+    no_copy_source = True
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+
+    def package(self):
+        self.copy(f'icecream-cpp-{self.version}/icecream.hpp', 'include/', keep_path=False)
```
Comment (madebr) [inline recipes/icecream-cpp/all/conanfile.py:15]: No f-strings (yet).
Older python versions are still supported.

File: recipes/icecream-cpp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+class IcecreamcppConan(ConanFile):
+    name = "icecream-cpp"
+    license = "MIT"
+    url = "https://github.com/renatoGarcia/icecream-cpp"
+    description = "A little library to help with the print debugging on C++11 and forward."
+    topics = ("debug", "single-header-lib", "print")
+    no_copy_source = True
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
```
Comment (madebr) [inline recipes/icecream-cpp/all/conanfile.py:12]: If you rename the source folder once here, you don't need to use version dependent variables in package.

File: recipes/icecream-cpp/all/conanfile.py (hunk 1 above)
Comment (renatoGarcia) [reply recipes/icecream-cpp/all/conanfile.py]: Ok. Fixed that.

File: recipes/icecream-cpp/all/conanfile.py (hunk 2 above)
Comment (renatoGarcia) [reply recipes/icecream-cpp/all/conanfile.py:12]: I'm sorry. I don't get what you mean by that.

Comment (madebr) [reply recipes/icecream-cpp/all/conanfile.py:12]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version])
        os.rename("icecream-cpp-{}".format(self.version), self._source_subfolder)
```
And add a `_source_subfolder = "source_subfoler"` member variable.
You can then use this variable in the `package` method.
That way, this recipe has the same format as other.

Comment (renatoGarcia) [reply recipes/icecream-cpp/all/conanfile.py:12]: Is that really needed? With your suggested changes I will be importing a new module (`os`), making a call to one more method (`os.rename`) and defining a new member variable (`_source_subfolder`). IMHO with that I will be adding more complexity to a code working just fine as it is, to no obvious gain.

Comment (madebr) [reply recipes/icecream-cpp/all/conanfile.py:12]: Import in python is negligible. `conan` overhead is much bigger and uses it itself.
Your packages will have the same expected format as all others, saving time in the future.

Comment (renatoGarcia) [reply recipes/icecream-cpp/all/conanfile.py:15]: I was not arguing about performance, but code complexity like in "understanding the code" complexity. But whatever, this code is so small, and if you say that is the common practice. I have made the changes.

Comment (madebr) [reply recipes/icecream-cpp/all/conanfile.py:15]: Thanks!
There is also complexity in diversity of coding styles. :smile:

Comment (renatoGarcia) [reply recipes/icecream-cpp/all/conanfile.py:15]: I concede that point. :smile:

**Comment by renatoGarcia** [comment]
Ok, I have meged the changes. @madebr and @uilianries, thaks for your guidance!

**Pull request merged**